src/realm/obj.hpp

```diff
@@ -351,6 +338,7 @@ class Obj {
     friend class Set;
     friend class Table;
     friend class Transaction;
+    friend class CollectionParent;
```

Comment:
This is because Obj has two roles: both as parent of a collection, but also just as an interface to a objects in general. In this case CollectionParent uses a private interface to set and remove backlinks on the objects that the collections points to.